### PR TITLE
Kotlin conversion preferences test

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Purpose / Description
Kotlin conversion : PreferencesTest.java -> PreferencesTest.kt

## Approach
1. Convert JAVA file to Kotlin
2. Do some small changes manually

## How Has This Been Tested?
Runs test as expected

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt [hunk 1]
```diff
@@ -13,60 +13,53 @@
...
+    fun setDayOffsetSetsConfig() {
+        col.changeSchedulerVer(2)
+        val preferences = instance
+        val offset = getDayOffset(col)
+        MatcherAssert.assertThat("Default offset should be 4", offset, Matchers.`is`(4))
+        preferences.setDayOffset(2)
+        MatcherAssert.assertThat("rollover config should be set to new value", col.get_config("rollover", 4.toInt()), Matchers.`is`(2))
     }
 
+    val instance: Preferences
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt:59]: `@KotlinCleanup("use scope function")`

**Review by david-allison — COMMENTED** [review]
Looks good - small requests

**Review by david-allison — CHANGES_REQUESTED** [review]
There needs to be 2 commits on a Kotlin Migration. Please squash the last two

Some feedback hasn't been addressed

**Comment by divyansh-dxn** [comment]
> 

I didn't get the using scope function. Use scope function for what?
Did you mean ?
```kotlin
     val instance: Preferences
            get() = Preferences().apply { this.attachBaseContext(targetContext) }

**Comment by david-allison** [comment]
> > 
> 
> I didn't get the using scope function. Use scope function for what? Did you mean ?
> 
> ```kotlin
>      val instance: Preferences
>             get() = Preferences().apply { this.attachBaseContext(targetContext) }
> ```

Add a `@KotlinCleanup` annotation for the user who performs the cleanup, to suggest they use a scope function here.

We shouldn't refactor the code in the conversion, as this makes it much harder to understand the history.

**Comment by divyansh-dxn** [comment]
> 

When I try to add this annotation to the field, it throws an error `This annotation is not applicable to target 'member property without backing field or delegate`. 
It needs to add `AnnotationTarget.PROPERTY` to KotlinCleanup.kt annotation class.
Even this is true for the getter method.
What should I do?

**Comment by david-allison** [comment]
Rebase your branch onto main: https://github.com/ankidroid/Anki-Android/pull/10544

File: AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt [hunk 2]
```diff
@@ -13,60 +13,53 @@
...
-    public void testDayOffsetExhaustive() {
-        Preferences preferences = getInstance();
-        for (int i = 0; i < 24; i++) {
-            preferences.setDayOffset(i);
-            assertThat(Preferences.getDayOffset(getCol()), is(i));
+    fun testDayOffsetExhaustive() {
+        val preferences = instance
+        for (i in 0..23) {
+            preferences.setDayOffset(i)
+            MatcherAssert.assertThat(getDayOffset(col), Matchers.`is`(i))
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt]: This comment is still pending

[the remark this answers is not in the document]
Comment (divyansh-dxn) [reply AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt]: You meant 
```kotlin
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.equalTo
```
right?

[the remark this answers is not in the document]
Comment (divyansh-dxn) [reply AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt]: It's my first conversion task, that's why not going that smooth  🥲🥲

**Review by david-allison — APPROVED** [review]
Looks great, thanks!

**Review by mikehardy — APPROVED** [review]
Nice!

**Comment by mikehardy** [comment]
Hi there! MARCH OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for [March 2022 submissions](https://github.com/ankidroid/Anki-Android/pulls?q=is%3Apr+merged%3A2022-03-01..2022-03-31)

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of March.

Please note that GSoC contributions are okay for this process. Our philosophy is that our users have donated to AnkiDroid for *all* contributions. The only PRs that will not go through the OpenCollective process are ones *directly* related to a GSoC project from a selected participant, since those receive a stipend from GSoC itself.

Please understand that this means the monthly budget we have for payments might be stretched a bit and the cap on payments-per-person may be lower, but we try to make our process as fair and transparent as possible, we just need your understanding.

Thanks!

**Pull request merged**